### PR TITLE
implement storage and database dao GDEV-349

### PR DESCRIPTION
**Pull request by KerstenBreuer**
Title for squash and merge:
```
implement storage and database dao (GDEV-349)
```

Description for squash and merge:
```
Added a database DAO interface for handling DRS object
entries in the database:
  - get DRS objects by id
  - register new DRS objects
  - unregister existing DRS objects by id
 A Prostgres-specific implementation of this DAO is provided.
The S3-specific object storage was imported from the
ghga_service_chassis_lib.
Moreover, request/reply models were refactored.
```

### CODE REVIEW

File: db_migration/alembic/versions/20211117_162939_7854bc67c695_simplify_drsobject.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column("drs_objects", sa.Column("external_id", sa.String(), nullable=False))
+    op.add_column("drs_objects", sa.Column("md5_checksum", sa.String(), nullable=False))
+    op.add_column(
+        "drs_objects", sa.Column("registration_date", sa.DateTime(), nullable=False)
```
Comment (MoritzHahn1337) [inline db_migration/alembic/versions/20211117_162939_7854bc67c695_simplify_drsobject.py:24]: ga4gh drs has a required "created_time" field for drs_objects, which I always assumed would be the creation time of the file on the researchers side. (https://ga4gh.github.io/data-repository-service-schemas/preview/release/drs-1.0.0/docs/)

Is this registration_date meant as a replacement for creation_time?

**Review by MoritzHahn1337 — APPROVED** [review]

File: db_migration/alembic/versions/20211117_162939_7854bc67c695_simplify_drsobject.py (hunk 1 above)
Comment (KerstenBreuer) [reply db_migration/alembic/versions/20211117_162939_7854bc67c695_simplify_drsobject.py:24]: I think the specs leave a lot time for interpretation:
```
Timestamp of content creation in RFC3339.
(This is the creation time of the underlying content, not of the JSON object.)
```
It's not clear whether they mean the creation time at the researcher side (that is something that we most likely cannot always get from the submitters) or the creation time in our system. The latter is roughly equivalent with the registration date/time to the DRS service. So I think we can use this definition as it is more convenient for us to implement.

Comment (KerstenBreuer) [reply db_migration/alembic/versions/20211117_162939_7854bc67c695_simplify_drsobject.py:24]: Would that be OK?

Comment (MoritzHahn1337) [reply db_migration/alembic/versions/20211117_162939_7854bc67c695_simplify_drsobject.py:24]: Fine with me.

**Review by MoritzHahn1337 — APPROVED** [review]

**Pull request merged**